polylogue/branching.py

```
@@ -0,0 +1,197 @@
+        seen: set[tuple[str, ...]] = set()
+        unique_paths: List[List[str]] = []
+        for path in all_paths:
+            key = tuple(path)
+            if key in seen:
+                continue
+            seen.add(key)
+            unique_paths.append(path)
+        all_paths = unique_paths
+
+    if not all_paths:
+        # No explicit leaves – treat roots as single-node branches.
```

Ruff (0.15.9)

[warning] Comment contains ambiguous `–` (EN DASH). Did you mean `-` (HYPHEN-MINUS)?

(RUF003)